config/v1/config.go

```diff
@@ -91,65 +95,80 @@ const (
+	InfraBucket        *string           `json:"infra_s3_bucket" yaml:"infra_s3_bucket"`
+	InfraDynamoTable   *string           `json:"infra_dynamo_db_table" yaml:"infra_dynamo_db_table"`
+	Kind               *ComponentKind    `json:"kind,omitempty" yaml:"kind,omitempty"`
+	ModuleSource       *string           `json:"module_source" yaml:"module_source"`
+	Owner              *string           `json:"owner" yaml:"owner"`
+	Project            *string           `json:"project" yaml:"project"`
+	TerraformVersion   *string           `json:"terraform_version" yaml:"terraform_version"`
+	TfLint             *TfLint           `json:"tflint,omitempty" yaml:"tflint,omitempty"`
 }
 
 // Plugins contains configuration around plugins
 type Plugins struct {
-	CustomPlugins      map[string]*plugins.CustomPlugin `json:"custom_plugins,omitempty"`
-	TerraformProviders map[string]*plugins.CustomPlugin `json:"terraform_providers,omitempty"`
+	CustomPlugins      map[string]*plugins.CustomPlugin `json:"custom_plugins,omitempty" yaml:"custom_plugins,omitempty"`
+	TerraformProviders map[string]*plugins.CustomPlugin `json:"terraform_providers,omitempty" yaml:"terraform_providers,omitempty"`
 }
 
 // Module is a module
 type Module struct {
-	TerraformVersion *string `json:"terraform_version,omitempty"`
+	TerraformVersion *string `json:"terraform_version,omitempty" yaml:"terraform_version,omitempty"`
 }
 
 type TravisCI struct {
-	Enabled        bool   `json:"enabled,omitempty"`
-	AWSIAMRoleName string `json:"aws_iam_role_name"`
-	TestBuckets    int    `json:"test_buckets"`
+	Enabled        bool   `json:"enabled,omitempty" yaml:"enabled,omitempty"`
+	AWSIAMRoleName string `json:"aws_iam_role_name" yaml:"aws_iam_role_name"`
+	TestBuckets    int    `json:"test_buckets" yaml:"test_buckets"`
 }
 
 type Config struct {
-	Accounts map[string]Account `json:"accounts"`
-	Defaults Defaults           `json:"defaults"`
-	Docker   bool               `json:"docker,omitempty"`
-	Envs     map[string]Env     `json:"envs"`
-	Modules  map[string]Module  `json:"modules"`
-	Plugins  Plugins            `json:"plugins,omitempty"`
-	TravisCI *TravisCI          `json:"travis_ci,omitempty"`
+	Accounts map[string]Account `json:"accounts" yaml:"accounts"`
+	Defaults Defaults           `json:"defaults" yaml:"defaults"`
+	Docker   bool               `json:"docker,omitempty" yaml:"docker,omitempty"`
+	Envs     map[string]Env     `json:"envs" yaml:"envs"`
+	Modules  map[string]Module  `json:"modules" yaml:"modules"`
+	Plugins  Plugins            `json:"plugins,omitempty" yaml:"plugins"`
+	TravisCI *TravisCI          `json:"travis_ci,omitempty" yaml:"travis_ci,omitempty"`
 }
 
 func ReadConfig(b []byte) (*Config, error) {
+	var e error
 	c := &Config{
 		Docker: true,
 	}
-	e := json.Unmarshal(b, c)
+
+	if IsJSON(b) {
+		e = json.Unmarshal(b, c)
+		logrus.Warn("JSON is deprecated, consider migrating to yaml")
+	} else {
+		e = yaml.Unmarshal(b, c)
+	}
 	if e != nil {
```

Comment:
looks like you can get rid of this branch since you're already returning https://github.com/chanzuckerberg/fogg/pull/298/files#diff-374f4b941ce2213d7811071542d264a6R164

Reply:
Removing it fails the nil tests because c is still initialized.